# Keyboard shortcuts

**Preciona H**

## Importar librerías

In [5]:
import pandas as pd
import numpy as np

## Leer Dataset

In [6]:
#Ruta donde se va a leer el dataset
path= 'Dataset/demanda-de-electricidad-datos-mensuales.csv'
#Se le asigna a una variable de python el dataset por medio de la funcion de Pandas "pd.read_csv()"
dataset= pd.read_csv(path)

### Primera exploración del dataset con pandas.
###### .head()
    la funcion head te muestra las 5 primeras filas de tu dataset.

In [3]:
dataset.head()

indice_tiempo  demanda_total  demanda_residencial  comercio_e_industria  \
0    2001-01-01        6866.26                  NaN                   NaN   
1    2001-02-01        6473.10                  NaN                   NaN   
2    2001-03-01        6952.30                  NaN                   NaN   
3    2001-04-01        6183.35                  NaN                   NaN   
4    2001-05-01        6668.94                  NaN                   NaN   

   grandes_usuarios  temperatura_promedio  potencia_maxima  ede_tucuman  \
0               NaN                25.432          13501.0          NaN   
1               NaN                25.596          14061.0          NaN   
2               NaN                22.852          13780.0          NaN   
3               NaN                18.317          12866.0          NaN   
4               NaN                14.306          12968.0          NaN   

   edelap_sa  edenor_distribuidor       ...         energia_de_catamarca_sa  \
0        NaN                  NaN       ...                             NaN   
1        NaN                  NaN       ...                             NaN   
2        NaN                  NaN       ...                             NaN   
3        NaN                  NaN       ...                             NaN   
4        NaN                  NaN       ...                             NaN   

   energia_de_entre_rios_sa  energia_de_mendoza_sa  \
0                       NaN                    NaN   
1                       NaN                    NaN   
2                       NaN                    NaN   
3                       NaN                    NaN   
4                       NaN                    NaN   

   energia_san_juan_sa_exedessa  epec_distribuidor  epen_distribuidor  \
0                           NaN                NaN                NaN   
1                           NaN                NaN                NaN   
2                           NaN                NaN                NaN   
3                           NaN                NaN                NaN   
4                           NaN                NaN                NaN   

   epesf_distribuidor  recursos_y_energia_formosa_sa  secheep  spse_santa_cruz  
0                 NaN                            NaN      NaN              NaN  
1                 NaN                            NaN      NaN              NaN  
2                 NaN                            NaN      NaN              NaN  
3                 NaN                            NaN      NaN              NaN  
4                 NaN                            NaN      NaN              NaN  

[5 rows x 30 columns]

## Columnas del Dataset

In [4]:
dataset.columns

Index(['indice_tiempo', 'demanda_total', 'demanda_residencial',
       'comercio_e_industria', 'grandes_usuarios', 'temperatura_promedio',
       'potencia_maxima', 'ede_tucuman', 'edelap_sa', 'edenor_distribuidor',
       'edesal_distribuidor', 'edestesa_emp_dist__este', 'edesur_distribuidor',
       'emp_de_energia_de_rio_negro_sa', 'emp_dist_energ_atlantica',
       'emp_de_energia_de_la_rioja_sa', 'emp_dist_energia_de_salta',
       'emp_electric_misiones_sa', 'empresa_dis_estero_sa',
       'empresa_jujenia_de_energia_sa', 'energia_de_catamarca_sa',
       'energia_de_entre_rios_sa', 'energia_de_mendoza_sa',
       'energia_san_juan_sa_exedessa', 'epec_distribuidor',
       'epen_distribuidor', 'epesf_distribuidor',
       'recursos_y_energia_formosa_sa', 'secheep', 'spse_santa_cruz'],
      dtype='object')

##### ¿Cuantas columnas tiene el data set?

In [5]:
len(dataset.columns)

30

##### ¿Cuales son las dimensiones del Dataset?

In [6]:
dataset.shape

(213, 30)

   Tenemos **213 Filas** y **30 Columnas**

## Tipos de columnas
#### .dtypes 
    Es un atributo del DataFrame que te da los tipos de datos por cada Columna

In [10]:
dataset.dtypes

indice_tiempo                      object
demanda_total                     float64
demanda_residencial               float64
comercio_e_industria              float64
grandes_usuarios                  float64
temperatura_promedio              float64
potencia_maxima                   float64
ede_tucuman                       float64
edelap_sa                         float64
edenor_distribuidor               float64
edesal_distribuidor               float64
edestesa_emp_dist__este           float64
edesur_distribuidor               float64
emp_de_energia_de_rio_negro_sa    float64
emp_dist_energ_atlantica          float64
emp_de_energia_de_la_rioja_sa     float64
emp_dist_energia_de_salta         float64
emp_electric_misiones_sa          float64
empresa_dis_estero_sa             float64
empresa_jujenia_de_energia_sa     float64
energia_de_catamarca_sa           float64
energia_de_entre_rios_sa          float64
energia_de_mendoza_sa             float64
energia_san_juan_sa_exedessa      

#### Que tipos de datos tiene Pandas
* **object**: Texto
* **float**: Numerico con decimales
* **int**: Numerico entero
* **bool**: True o False, Valores binarios.
* **category**: Valores categoricos
* **datetime ó timedelta**: Son Datos con fechas o tiempos

## Aplicamos slicing solo de los datos generales y de la compañía Epec

In [12]:
dataset_epec = dataset[['indice_tiempo','demanda_residencial','comercio_e_industria','grandes_usuarios','temperatura_promedio','epec_distribuidor']]
dataset_epec

indice_tiempo  demanda_residencial  comercio_e_industria  \
0      2001-01-01                  NaN                   NaN   
1      2001-02-01                  NaN                   NaN   
2      2001-03-01                  NaN                   NaN   
3      2001-04-01                  NaN                   NaN   
4      2001-05-01                  NaN                   NaN   
5      2001-06-01                  NaN                   NaN   
6      2001-07-01                  NaN                   NaN   
7      2001-08-01                  NaN                   NaN   
8      2001-09-01                  NaN                   NaN   
9      2001-10-01                  NaN                   NaN   
10     2001-11-01                  NaN                   NaN   
11     2001-12-01                  NaN                   NaN   
12     2002-01-01                  NaN                   NaN   
13     2002-02-01                  NaN                   NaN   
14     2002-03-01                  NaN                   NaN   
15     2002-04-01                  NaN                   NaN   
16     2002-05-01                  NaN                   NaN   
17     2002-06-01                  NaN                   NaN   
18     2002-07-01                  NaN                   NaN   
19     2002-08-01                  NaN                   NaN   
20     2002-09-01                  NaN                   NaN   
21     2002-10-01                  NaN                   NaN   
22     2002-11-01                  NaN                   NaN   
23     2002-12-01                  NaN                   NaN   
24     2003-01-01                  NaN                   NaN   
25     2003-02-01                  NaN                   NaN   
26     2003-03-01                  NaN                   NaN   
27     2003-04-01                  NaN                   NaN   
28     2003-05-01                  NaN                   NaN   
29     2003-06-01                  NaN                   NaN   
..            ...                  ...                   ...   
183    2016-04-01          4104.173856           4148.827000   
184    2016-05-01          4941.165168           4359.846460   
185    2016-06-01          5634.190447           4282.756603   
186    2016-07-01          5614.534651           4366.110724   
187    2016-08-01          4663.796035           4249.645917   
188    2016-09-01          4351.181774           4063.899360   
189    2016-10-01          3953.447552           3985.591323   
190    2016-11-01          3970.327674           4142.212910   
191    2016-12-01          5148.817670           4687.036718   
192    2017-01-01          5681.816626           4849.049020   
193    2017-02-01          4618.226100           4551.680399   
194    2017-03-01          4470.715122           4514.516433   
195    2017-04-01          3754.330119           3986.428466   
196    2017-05-01          4524.024791           4137.188624   
197    2017-06-01          5180.585335           4149.745327   
198    2017-07-01          5460.615925           4259.809832   
199    2017-08-01          4767.354968           4209.895737   
200    2017-09-01          4217.808567           4048.775981   
201    2017-10-01          3979.758234           4117.571079   
202    2017-11-01          4035.416488           4174.310998   
203    2017-12-01          5216.357135           4704.238326   
204    2018-01-01          5468.423592           4781.759728   
205    2018-02-01          4963.781739           4416.421143   
206    2018-03-01          4492.917162           4579.324971   
207    2018-04-01          4084.109752           4269.204833   
208    2018-05-01          4276.829777           4236.631730   
209    2018-06-01          5569.775499           4332.322478   
210    2018-07-01          6167.113772           4421.727938   
211    2018-08-01          5359.054186           4301.646845   
212    2018-09-01          3995.352693           3815.260592   

     grandes_usuarios  temperatura_

### Podemos buscar solo los datos que sean del 2018
###### Con la funcion de Pandas .loc

In [13]:
(dataset_epec.indice_tiempo >= '2018') 

0      False
1      False
2      False
3      False
4      False
5      False
6      False
7      False
8      False
9      False
10     False
11     False
12     False
13     False
14     False
15     False
16     False
17     False
18     False
19     False
20     False
21     False
22     False
23     False
24     False
25     False
26     False
27     False
28     False
29     False
       ...  
183    False
184    False
185    False
186    False
187    False
188    False
189    False
190    False
191    False
192    False
193    False
194    False
195    False
196    False
197    False
198    False
199    False
200    False
201    False
202    False
203    False
204     True
205     True
206     True
207     True
208     True
209     True
210     True
211     True
212     True
Name: indice_tiempo, Length: 213, dtype: bool

In [19]:
dataset_epec.sum()

indice_tiempo           2001-01-012001-02-012001-03-012001-04-012001-0...
demanda_residencial                                                627866
comercio_e_industria                                               643739
grandes_usuarios                                                   331209
temperatura_promedio                                              3927.17
epec_distribuidor                                                 42293.1
dtype: object

In [16]:
dataset_epec.loc[dataset_epec.demanda_residencial > 3000]

indice_tiempo  demanda_residencial  comercio_e_industria  \
67     2006-08-01          3107.201000           3472.059000   
71     2006-12-01          3071.673000           3802.947000   
72     2007-01-01          3079.471000           3735.829000   
76     2007-05-01          3161.771000           3731.939000   
77     2007-06-01          3576.113000           3535.967000   
78     2007-07-01          3807.798000           3564.032000   
79     2007-08-01          3726.224000           3589.396000   
83     2007-12-01          3138.571000           3815.519000   
84     2008-01-01          3420.283000           3811.277000   
85     2008-02-01          3073.664000           3705.406000   
88     2008-05-01          3102.396000           3625.434000   
89     2008-06-01          3699.403000           3582.687000   
90     2008-07-01          3626.348000           3476.582000   
91     2008-08-01          3438.115000           3488.405000   
92     2008-09-01          3061.343000           3375.917000   
94     2008-11-01          3159.108000           3681.432000   
95     2008-12-01          3471.033000           3691.127000   
96     2009-01-01          3533.299000           3658.961000   
97     2009-02-01          3105.440000           3404.180000   
98     2009-03-01          3336.438000           3769.532000   
100    2009-05-01          3047.803000           3510.857000   
101    2009-06-01          3721.346000           3529.784000   
102    2009-07-01          4052.004000           3612.156000   
103    2009-08-01          3502.447000           3400.943000   
104    2009-09-01          3023.058000           3474.772000   
105    2009-10-01          3027.540000           3471.550000   
106    2009-11-01          3065.794000           3643.166000   
107    2009-12-01          3336.758000           3780.482000   
108    2010-01-01          3931.898000           3934.752000   
109    2010-02-01          3361.184000           3579.086000   
..            ...                  ...                   ...   
183    2016-04-01          4104.173856           4148.827000   
184    2016-05-01          4941.165168           4359.846460   
185    2016-06-01          5634.190447           4282.756603   
186    2016-07-01          5614.534651           4366.110724   
187    2016-08-01          4663.796035           4249.645917   
188    2016-09-01          4351.181774           4063.899360   
189    2016-10-01          3953.447552           3985.591323   
190    2016-11-01          3970.327674           4142.212910   
191    2016-12-01          5148.817670           4687.036718   
192    2017-01-01          5681.816626           4849.049020   
193    2017-02-01          4618.226100           4551.680399   
194    2017-03-01          4470.715122           4514.516433   
195    2017-04-01          3754.330119           3986.428466   
196    2017-05-01          4524.024791           4137.188624   
197    2017-06-01          5180.585335           4149.745327   
198    2017-07-01          5460.615925           4259.809832   
199    2017-08-01          4767.354968           4209.895737   
200    2017-09-01          4217.808567           4048.775981   
201    2017-10-01          3979.758234           4117.571079   
202    2017-11-01          4035.416488           4174.310998   
203    2017-12-01          5216.357135           4704.238326   
204    2018-01-01          5468.423592           4781.759728   
205    2018-02-01          4963.781739           4416.421143   
206    2018-03-01          4492.917162           4579.324971   
207    2018-04-01          4084.109752           4269.204833   
208    2018-05-01          4276.829777           4236.631730   
209    2018-06-01          5569.775499           4332.322478   
210    2018-07-01          6167.113772           4421.727938   
211    2018-08-01          5359.054186           4301.646845   
212    2018-09-01          3995.352693           3815.260592   

     grandes_usuarios  temperatura_

## Clase que viene Empezaremos de lleno con Pandas

# Ya empezamos con Pandas

In [7]:
dataset.isnull().sum()

indice_tiempo                       0
demanda_total                       0
demanda_residencial                48
comercio_e_industria               48
grandes_usuarios                   48
temperatura_promedio                0
potencia_maxima                     0
ede_tucuman                       156
edelap_sa                         156
edenor_distribuidor               156
edesal_distribuidor               156
edestesa_emp_dist__este           156
edesur_distribuidor               156
emp_de_energia_de_rio_negro_sa    156
emp_dist_energ_atlantica          156
emp_de_energia_de_la_rioja_sa     156
emp_dist_energia_de_salta         156
emp_electric_misiones_sa          156
empresa_dis_estero_sa             156
empresa_jujenia_de_energia_sa     156
energia_de_catamarca_sa           156
energia_de_entre_rios_sa          156
energia_de_mendoza_sa             156
energia_san_juan_sa_exedessa      156
epec_distribuidor                 156
epen_distribuidor                 156
epesf_distri

In [9]:
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

In [11]:
dataset.head(200)

indice_tiempo  demanda_total  demanda_residencial  comercio_e_industria  \
0      2001-01-01    6866.260000                  NaN                   NaN   
1      2001-02-01    6473.100000                  NaN                   NaN   
2      2001-03-01    6952.300000                  NaN                   NaN   
3      2001-04-01    6183.350000                  NaN                   NaN   
4      2001-05-01    6668.940000                  NaN                   NaN   
5      2001-06-01    6559.160000                  NaN                   NaN   
6      2001-07-01    6806.800000                  NaN                   NaN   
7      2001-08-01    6531.180000                  NaN                   NaN   
8      2001-09-01    6211.660000                  NaN                   NaN   
9      2001-10-01    6268.030000                  NaN                   NaN   
10     2001-11-01    6132.660000                  NaN                   NaN   
11     2001-12-01    6448.414000                  NaN                   NaN   
12     2002-01-01    6471.690000                  NaN                   NaN   
13     2002-02-01    5745.320000                  NaN                   NaN   
14     2002-03-01    6462.130000                  NaN                   NaN   
15     2002-04-01    5970.250000                  NaN                   NaN   
16     2002-05-01    6320.570000                  NaN                   NaN   
17     2002-06-01    6634.900000                  NaN                   NaN   
18     2002-07-01    6891.220000                  NaN                   NaN   
19     2002-08-01    6514.580000                  NaN                   NaN   
20     2002-09-01    6136.420000                  NaN                   NaN   
21     2002-10-01    6344.780000                  NaN                   NaN   
22     2002-11-01    6384.030000                  NaN                   NaN   
23     2002-12-01    6597.000000                  NaN                   NaN   
24     2003-01-01    7009.410000                  NaN                   NaN   
25     2003-02-01    6364.910000                  NaN                   NaN   
26     2003-03-01    6874.260000                  NaN                   NaN   
27     2003-04-01    6280.380000                  NaN                   NaN   
28     2003-05-01    6773.800000                  NaN                   NaN   
29     2003-06-01    6814.530000                  NaN                   NaN   
30     2003-07-01    7321.710000                  NaN                   NaN   
31     2003-08-01    7188.130000                  NaN                   NaN   
32     2003-09-01    6737.750000                  NaN                   NaN   
33     2003-10-01    6906.960000                  NaN                   NaN   
34     2003-11-01    6833.970000                  NaN                   NaN   
35     2003-12-01    7107.560000                  NaN                   NaN   
36     2004-01-01    7620.560000                  NaN                   NaN   
37     2004-02-01    6846.540000                  NaN                   NaN   
38     2004-03-01    7544.990000                  NaN                   NaN   
39     2004-04-01    7011.910000                  NaN                   NaN   
40     2004-05-01    7325.250000                  NaN                   NaN   
41     2004-06-01    7259.350000                  NaN                   NaN   
42     2004-07-01    7576.600000                  NaN                   NaN   
43     2004-08-01    7076.070000                  NaN                   NaN   
44     2004-09-01    6956.970000                  NaN                   NaN   
45     2004-10-01    7087.330000                  NaN                   NaN   
46     2004-11-01    7127.320000                  NaN                   NaN   
47     2004-12-01    7739.520000                  NaN                   NaN   
48     2005-01-01    7983.280000          2717.678000           3750.612000   
49     2005-02-01    7248.880000          2298.744000 

In [23]:
promedios_columns = dataset.describe().loc['mean']

In [25]:
promedios_columns.to_dict()

{'comercio_e_industria': 3902.2165578454824,
 'demanda_residencial': 3805.619170003915,
 'demanda_total': 9049.776228144863,
 'ede_tucuman': 228.51142772413786,
 'edelap_sa': 215.10572967241387,
 'edenor_distribuidor': 1841.9087888965516,
 'edesal_distribuidor': 104.66923041379313,
 'edestesa_emp_dist__este': 63.17505856896552,
 'edesur_distribuidor': 1482.1748207931037,
 'emp_de_energia_de_la_rioja_sa': 100.87039668965518,
 'emp_de_energia_de_rio_negro_sa': 110.44993160344829,
 'emp_dist_energ_atlantica': 226.89051793103442,
 'emp_dist_energia_de_salta': 157.22311599999995,
 'emp_electric_misiones_sa': 169.80589996551723,
 'empresa_dis_estero_sa': 118.52354005172415,
 'empresa_jujenia_de_energia_sa': 75.5675031724138,
 'energia_de_catamarca_sa': 81.26550110344827,
 'energia_de_entre_rios_sa': 231.6851751551725,
 'energia_de_mendoza_sa': 279.1930417586207,
 'energia_san_juan_sa_exedessa': 145.99397427586206,
 'epec_distribuidor': 752.1713190862067,
 'epen_distribuidor': 76.659068931034

In [21]:
dataset.fillna?

In [36]:
dataset_sin_null= dataset.fillna(promedios_columns)

In [38]:
promedios_columns?

In [37]:
dataset_sin_null

indice_tiempo  demanda_total  demanda_residencial  comercio_e_industria  \
0      2001-01-01    6866.260000          3805.619170           3902.216558   
1      2001-02-01    6473.100000          3805.619170           3902.216558   
2      2001-03-01    6952.300000          3805.619170           3902.216558   
3      2001-04-01    6183.350000          3805.619170           3902.216558   
4      2001-05-01    6668.940000          3805.619170           3902.216558   
5      2001-06-01    6559.160000          3805.619170           3902.216558   
6      2001-07-01    6806.800000          3805.619170           3902.216558   
7      2001-08-01    6531.180000          3805.619170           3902.216558   
8      2001-09-01    6211.660000          3805.619170           3902.216558   
9      2001-10-01    6268.030000          3805.619170           3902.216558   
10     2001-11-01    6132.660000          3805.619170           3902.216558   
11     2001-12-01    6448.414000          3805.619170           3902.216558   
12     2002-01-01    6471.690000          3805.619170           3902.216558   
13     2002-02-01    5745.320000          3805.619170           3902.216558   
14     2002-03-01    6462.130000          3805.619170           3902.216558   
15     2002-04-01    5970.250000          3805.619170           3902.216558   
16     2002-05-01    6320.570000          3805.619170           3902.216558   
17     2002-06-01    6634.900000          3805.619170           3902.216558   
18     2002-07-01    6891.220000          3805.619170           3902.216558   
19     2002-08-01    6514.580000          3805.619170           3902.216558   
20     2002-09-01    6136.420000          3805.619170           3902.216558   
21     2002-10-01    6344.780000          3805.619170           3902.216558   
22     2002-11-01    6384.030000          3805.619170           3902.216558   
23     2002-12-01    6597.000000          3805.619170           3902.216558   
24     2003-01-01    7009.410000          3805.619170           3902.216558   
25     2003-02-01    6364.910000          3805.619170           3902.216558   
26     2003-03-01    6874.260000          3805.619170           3902.216558   
27     2003-04-01    6280.380000          3805.619170           3902.216558   
28     2003-05-01    6773.800000          3805.619170           3902.216558   
29     2003-06-01    6814.530000          3805.619170           3902.216558   
30     2003-07-01    7321.710000          3805.619170           3902.216558   
31     2003-08-01    7188.130000          3805.619170           3902.216558   
32     2003-09-01    6737.750000          3805.619170           3902.216558   
33     2003-10-01    6906.960000          3805.619170           3902.216558   
34     2003-11-01    6833.970000          3805.619170           3902.216558   
35     2003-12-01    7107.560000          3805.619170           3902.216558   
36     2004-01-01    7620.560000          3805.619170           3902.216558   
37     2004-02-01    6846.540000          3805.619170           3902.216558   
38     2004-03-01    7544.990000          3805.619170           3902.216558   
39     2004-04-01    7011.910000          3805.619170           3902.216558   
40     2004-05-01    7325.250000          3805.619170           3902.216558   
41     2004-06-01    7259.350000          3805.619170           3902.216558   
42     2004-07-01    7576.600000          3805.619170           3902.216558   
43     2004-08-01    7076.070000          3805.619170           3902.216558   
44     2004-09-01    6956.970000          3805.619170           3902.216558   
45     2004-10-01    7087.330000          3805.619170           3902.216558   
46     2004-11-01    7127.320000          3805.619170           3902.216558   
47     2004-12-01    7739.520000          3805.619170           3902.216558   
48     2005-01-01    7983.280000          2717.678000           3750.612000   
49     2005-02-01    7248.880000          2298.744000 

In [29]:
serie_fechas = dataset.indice_tiempo

In [32]:
dataframe_tiempo = dataset[['indice_tiempo']]

In [39]:
dataset.fillna?

In [34]:
dataframe_tiempo['nueva columna'] = dataset_sin_null['demanda_total']

/home/arios/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


In [35]:
dataframe_tiempo

indice_tiempo  nueva columna
0      2001-01-01    6866.260000
1      2001-02-01    6473.100000
2      2001-03-01    6952.300000
3      2001-04-01    6183.350000
4      2001-05-01    6668.940000
5      2001-06-01    6559.160000
6      2001-07-01    6806.800000
7      2001-08-01    6531.180000
8      2001-09-01    6211.660000
9      2001-10-01    6268.030000
10     2001-11-01    6132.660000
11     2001-12-01    6448.414000
12     2002-01-01    6471.690000
13     2002-02-01    5745.320000
14     2002-03-01    6462.130000
15     2002-04-01    5970.250000
16     2002-05-01    6320.570000
17     2002-06-01    6634.900000
18     2002-07-01    6891.220000
19     2002-08-01    6514.580000
20     2002-09-01    6136.420000
21     2002-10-01    6344.780000
22     2002-11-01    6384.030000
23     2002-12-01    6597.000000
24     2003-01-01    7009.410000
25     2003-02-01    6364.910000
26     2003-03-01    6874.260000
27     2003-04-01    6280.380000
28     2003-05-01    6773.800000
29     2003-06-01    6814.530000
30     2003-07-01    7321.710000
31     2003-08-01    7188.130000
32     2003-09-01    6737.750000
33     2003-10-01    6906.960000
34     2003-11-01    6833.970000
35     2003-12-01    7107.560000
36     2004-01-01    7620.560000
37     2004-02-01    6846.540000
38     2004-03-01    7544.990000
39     2004-04-01    7011.910000
40     2004-05-01    7325.250000
41     2004-06-01    7259.350000
42     2004-07-01    7576.600000
43     2004-08-01    7076.070000
44     2004-09-01    6956.970000
45     2004-10-01    7087.330000
46     2004-11-01    7127.320000
47     2004-12-01    7739.520000
48     2005-01-01    7983.280000
49     2005-02-01    7248.880000
50     2005-03-01    7689.910000
51     2005-04-01    7213.140000
52     2005-05-01    7668.550000
53     2005-06-01    7719.710000
54     2005-07-01    8061.620000
55     2005-08-01    7953.500000
56     2005-09-01    7508.400000
57     2005-10-01    7403.310000
58     2005-11-01    7784.340000
59     2005-12-01    8098.310000
60     2006-01-01    7957.900000
61     2006-02-01    7395.700000
62     2006-03-01    8037.600000
63     2006-04-01    7555.600000
64     2006-05-01    8116.100000
65     2006-06-01    8248.900000
66     2006-07-01    8313.500000
67     2006-08-01    8497.000000
68     2006-09-01    7828.400000
69     2006-10-01    8025.100000
70     2006-11-01    7998.100000
71     2006-12-01    8838.900000
72     2007-01-01    8743.200000
73     2007-02-01    8158.200000
74     2007-03-01    8637.800000
75     2007-04-01    8010.400000
76     2007-05-01    8789.500000
77     2007-06-01    8851.400000
78     2007-07-01    9024.500000
79     2007-08-01    9106.500000
80     2007-09-01    8082.600000
81     2007-10-01    8370.400000
82     2007-11-01    8254.400000
83     2007-12-01    8921.400000
84     2008-01-01    9175.100000
85     2008-02-01    8728.800000
86     2008-03-01    8594.500000
87     2008-04-01    8286.600000
88     2008-05-01    8777.800000
89     2008-06-01    9209.800000
90     2008-07-01    9127.100000
91     2008-08-01    9025.300000
92     2008-09-01    8587.600000
93     2008-10-01    8477.200000
94     2008-11-01    8914.900000
95     2008-12-01    9053.400000
96     2009-01-01    9086.100000
97     2009-02-01    8241.600000
98     2009-03-01    8955.600000
99     2009-04-01    8104.700000
100    2009-05-01    8435.300000
101    2009-06-01    9114.200000
102    2009-07-01    9508.000000
103    2009-08-01    8738.400000
104    2009-09-01    8350.200000
105    2009-10-01    8420.700000
106    2009-11-01    8644.000000
107    2009-12-01    9023.600000
108    2010-01-01    9740.000000
109    2010-02-01    8773.900000
110    2010-03-01    9518.900000
111    2010-04-01    8434.100000
112    2010-05-01    8866.400000
113    2010-06-01    9367.000000
114    2010-07-01   10081.100000
115    2010-08-01    9733.100000
116    2010-09-01    8755.900000
117    2010-10-01    8567.700000
118    2010-11-01    8834.600000
119    2010-12-01   10125.700000
120    2011